packages/graphql-testing/src/MemoryApolloClient.ts

```diff
@@ -0,0 +1,45 @@
+import ApolloClient, {ApolloClientOptions} from 'apollo-client';
+import {ApolloLink} from 'apollo-link';
+import {Omit} from '@shopify/useful-types';
+
+import MemoryApolloLink from './MemoryApolloLink';
+import Operations from './Operations';
+import {MockRequest} from './types';
+
+type MemoryApolloClientOptions = Omit<ApolloClientOptions<unknown>, 'link'> & {
+  link?: ApolloLink;
+};
+
+export default class MemoryApolloClient extends ApolloClient<unknown> {
+  readonly operations: Operations;
+  readonly pendingRequests: Set<MockRequest>;
+
+  constructor(options: MemoryApolloClientOptions) {
+    const memoryLink = new MemoryApolloLink();
+    super({
+      ...options,
+      link: options.link ? memoryLink.concat(options.link) : memoryLink,
+    });
+
+    this.operations = new Operations();
+    this.pendingRequests = new Set<MockRequest>();
+
+    memoryLink.requestCreated(this.requestCreated.bind(this));
```

Comment:
I feel like these should maybe be prefixed with `on`, they feel a little awkward right now